presto-orc/src/test/java/com/facebook/presto/orc/metadata/TestDwrfMetadataReader.java

```diff
@@ -0,0 +1,187 @@
+                                    null,
+                                    getMaxSlice(testValue.toStringUtf8()),
+                                    79));
+                }
+            }
+        }
+
+        // Test truncation of new version due to replacement of invalid UTF-8 sequences
+        for (Slice prefix : TEST_SEQUENCES) {
+            for (int testCodePoint : TEST_CODE_POINTS) {
+                Slice codePoint = codePointToUtf8(testCodePoint);
+                for (Slice suffix : TEST_SEQUENCES) {
+                    Slice testValue = concatSlice(prefix, codePoint, suffix);
+                    assertEquals(
+                            DwrfMetadataReader.toStringStatistics(
+                                    HiveWriterVersion.ORC_HIVE_8732,
+                                    DwrfProto.StringStatistics.newBuilder()
+                                            .setMinimumBytes(ByteString.copyFrom(testValue.getBytes()))
+                                            .setMaximumBytes(ByteString.copyFrom(testValue.getBytes()))
+                                            .setSum(79)
+                                            .build(),
+                                    true),
+                            new StringStatistics(
+                                    minStringTruncateAtFirstReplacementCharacter(testValue),
+                                    maxStringTruncateAtFirstReplacementCharacter(testValue),
+                                    79));
+                    assertEquals(
+                            DwrfMetadataReader.toStringStatistics(
+                                    HiveWriterVersion.ORC_HIVE_8732,
+                                    DwrfProto.StringStatistics.newBuilder()
+                                            .setMinimumBytes(ByteString.copyFrom(testValue.getBytes()))
+                                            .setSum(79)
+                                            .build(),
+                                    true),
+                            new StringStatistics(
+                                    minStringTruncateAtFirstReplacementCharacter(testValue),
+                                    null,
+                                    79));
+                    assertEquals(
+                            DwrfMetadataReader.toStringStatistics(
+                                    HiveWriterVersion.ORC_HIVE_8732,
+                                    DwrfProto.StringStatistics.newBuilder()
+                                            .setMaximumBytes(ByteString.copyFrom(testValue.getBytes()))
+                                            .setSum(79)
+                                            .build(),
+                                    true),
+                            new StringStatistics(
+                                    null,
+                                    maxStringTruncateAtFirstReplacementCharacter(testValue),
+                                    79));
```

Comment:
I would argue for
1. making these three `assertEquals` a function called `testTruncationOfReplacementCharacter`, and the three `assertEquals` earlier a function called `testTruncationOfSupplementCharacter`. 
2. Make both functions take a version
3. Merge the two 3-level for loop 

This would make the code express what you wrote in the comment.

If you agree with me that replacement character truncation should also apply to ORIGINAL, this would have the additional benefit of easily applying `testTruncationOfReplacementCharacter` to ORIGINAL.